src/theory/datatypes/infer_proof_cons.cpp

```diff
@@ -219,18 +219,33 @@ void InferProofCons::convert(InferenceId infer, TNode conc, TNode exp, CDProof*
     {
       Assert(2 <= expv.size() && expv.size() <= 3);
       Node tester1 = expv[0];
-      Node tester1c =
-          nm->mkNode(APPLY_TESTER, expv[1].getOperator(), expv[0][0]);
-      std::vector<Node> targs{expv[1]};
-      if (expv.size() == 3)
+      bool pol = expv[1].getKind() != NOT;
```

Comment:
Can you take the opportunity and add documentation describing what is the shape of the proof being built here?